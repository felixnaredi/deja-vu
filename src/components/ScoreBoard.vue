<template>
  <div class="grid grid-cols-3 mx-10">
    <div class="justify-self-center w-1/2">
      <span class="mr-1">score:</span>
      <span>{{ score }}</span>
    </div>
    <div></div>
    <div class="justify-self-center w-1/2">
      <span class="mr-1">des vies:</span>
      <span>{{ lives }}</span>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    score: Number,
    lives: Number,
  },
};
</script>
