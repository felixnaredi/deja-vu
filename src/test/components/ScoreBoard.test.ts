import ScoreBoard from "@/components/ScoreBoard.vue";
import { mount } from "@vue/test-utils";

describe("unit tests", () => {
  test("empty render accordingly", () => {
    const wrapper = mount(ScoreBoard);
    const columns = wrapper.find("div").findAll("div");
    for (const { i, label } of [
      { i: 0, label: "score:" },
      { i: 2, label: "des vies:" },
    ]) {
      const spans = columns[i].findAll("span");
      expect(spans[0].text()).toBe(label);
      expect(spans[1].text()).toBe("");
    }
  });
  test("renders accordingly when score and lives prop changes", async () => {
    const wrapper = mount(ScoreBoard);
    for (const props of [
      { score: 0, lives: 0 },
      { score: 1, lives: 3 },
      { score: 7312, lives: -8312 },
      { score: -62, lives: 81 },
    ]) {
      await wrapper.setProps(props);
      const { score, lives } = props;
      const columns = wrapper.find("div").findAll("div");
      expect(columns[0].findAll("span")[1].text()).toBe(String(score));
      expect(columns[2].findAll("span")[1].text()).toBe(String(lives));
    }
  });
});

describe("snapshots", () => {
  test("empty", () => {
    expect(mount(ScoreBoard).html()).toMatchSnapshot();
  });
  test("score and lives prop changes", async () => {
    const wrapper = mount(ScoreBoard);
    for (const props of [
      { score: 0, lives: 0 },
      { score: 1, lives: 3 },
      { score: 7312, lives: -8312 },
      { score: -62, lives: 81 },
    ]) {
      await wrapper.setProps(props);
      expect(wrapper.html()).toMatchSnapshot();
    }
  });
});
