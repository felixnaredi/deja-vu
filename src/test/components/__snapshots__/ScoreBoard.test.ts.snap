// Vitest Snapshot v1

exports[`snapshots > empty 1`] = `
"<div class=\\"grid grid-cols-3 mx-10\\">
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">score:</span><span></span></div>
  <div></div>
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">des vies:</span><span></span></div>
</div>"
`;

exports[`snapshots > score and lives prop changes 1`] = `
"<div class=\\"grid grid-cols-3 mx-10\\">
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">score:</span><span>0</span></div>
  <div></div>
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">des vies:</span><span>0</span></div>
</div>"
`;

exports[`snapshots > score and lives prop changes 2`] = `
"<div class=\\"grid grid-cols-3 mx-10\\">
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">score:</span><span>1</span></div>
  <div></div>
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">des vies:</span><span>3</span></div>
</div>"
`;

exports[`snapshots > score and lives prop changes 3`] = `
"<div class=\\"grid grid-cols-3 mx-10\\">
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">score:</span><span>7312</span></div>
  <div></div>
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">des vies:</span><span>-8312</span></div>
</div>"
`;

exports[`snapshots > score and lives prop changes 4`] = `
"<div class=\\"grid grid-cols-3 mx-10\\">
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">score:</span><span>-62</span></div>
  <div></div>
  <div class=\\"justify-self-center w-1/2\\"><span class=\\"mr-1\\">des vies:</span><span>81</span></div>
</div>"
`;
